pkg/abap/build/connector.go

```diff
@@ -174,3 +176,39 @@ func (conn Connector) UploadSarFile(appendum string, sarFile []byte) error {
 	defer response.Body.Close()
 	return nil
 }
+
+// UploadSarFileInChunks : upload *.sar file in chunks
+func (conn Connector) UploadSarFileInChunks(appendum string, fileName string, sarFile []byte) error {
+	//Maybe Next Refactoring step to read the file in chunks, too?
+
+	url := conn.Baseurl + appendum
+
+	header := make(map[string][]string)
+	header["Content-Disposition"] = []string{"form-data; name=\"file\"; filename=\"" + fileName + "\""}
+
+	//chunkSize := 10000 // 10KB for testing
+	//chunkSize := 1000000 //1MB for Testing,
+	chunkSize := 10000000 //10MB
+	log.Entry().Infof("Upload in chunks of %d bytes", chunkSize)
+
+	sarFileBuffer := bytes.NewBuffer(sarFile)
+	fileSize := sarFileBuffer.Len()
+
+	for sarFileBuffer.Len() > 0 {
+		startOffset := fileSize - sarFileBuffer.Len()
+		nextChunk := bytes.NewBuffer(sarFileBuffer.Next(chunkSize))
+		endOffset := fileSize - sarFileBuffer.Len()
+		header["Content-Range"] = []string{"bytes " + strconv.Itoa(startOffset) + " - " + strconv.Itoa(endOffset) + " / " + strconv.Itoa(fileSize)}
+		log.Entry().Info(header["Content-Range"])
+
+		response, err := conn.Client.SendRequest("POST", url, nextChunk, header, nil)
+		if err != nil {
+			errorbody, _ := ioutil.ReadAll(response.Body)
+			defer response.Body.Close()
+			return errors.Wrapf(err, "Upload of SAR file failed: %v", string(errorbody))
```

Comment:
IMHO, for it to make more sense to upload the file in chunks, you would have to implement some kind of retry mechanism. Perhaps based on the actual http status, but that can also be left for later.

Reply:
Reason for upload in chunks is that due to security reasons (e.g. denial of service aspect) systems usually limit the upload. If you want to upload beyond that limit it hast to be done in chunks. retry mechanism does not make too much sense for me as we are in the protocol stack already on top of TCP which provides error detection and correction. If this does not work it will most likely also not work in some seconds.

Reply:
> Reason for upload in chunks is that due to security reasons (e.g. denial of service aspect) systems usually limit the upload. If you want to upload beyond that limit it hast to be done in chunks.

Could have been part of the PR description.

> retry mechanism does not make too much sense for me as we are in the protocol stack already on top of TCP which provides error detection and correction. If this does not work it will most likely also not work in some seconds.

It sounds like it makes sense. But I am not convinced that retrying would not catch some situations where the error correction of the protocol layer doesn't help. I hope this sentence makes sense with a total of three negations. Let's try it the other way around: I am convinced that retries would catch some situations where the protocol layer's error correction doesn't help. But I see how that is not the scope of your PR, you just want to work around the upload size limit.

Reply:
added the idea as comment in the method